cylc/flow/data_store_mgr.py

```diff
@@ -1665,6 +1654,7 @@ def insert_job(
             execution_time_limit=job_conf.get('execution_time_limit'),
             platform=job_conf['platform']['name'],
             job_runner_name=job_conf.get('job_runner_name'),
+            job_id=itask.summary.get('submit_method_id'),
```

Comment:
Note, there is no Job ID for a submission failure caused by a platform lookup error because no job submission was made.

Reply:
`.get()` will return `None` in this case, which is fine